src/Poseidon/Janno.hs

```diff
@@ -289,6 +305,34 @@ instance ToJSON JannoLibraryBuilt where
 instance FromJSON JannoLibraryBuilt --where
     --parseJSON = withText "JannoLibraryBuilt" (makeJannoLibraryBuilt . T.unpack)
 
+-- | A datatype for countries in ISO-alpha2 code format
+newtype JannoCountry = JannoCountry Country
+    deriving (Eq, Ord)
+
+instance Show JannoCountry where
+    show (JannoCountry x) = T.unpack $ alphaTwoUpper x
+
+makeJannoCountryUnsafe :: String -> JannoCountry
```

Comment:
OK, I just realized that `MonadFail` is a pattern used in the JSON and CSV instances, also in Janno, and if that's the thing that's needed, OK. But generally, if we want a monad that throws exceptions, we should really be using `MonadThrow` I think.

Reply:
I think we ran into the problem raised in the blog post I shared with you one month ago: https://alexey.kuleshevi.ch/blog/2023/01/16/fail-with-class

> It seems that we can always use mkHour to safely construct an Hour from a variety of inputs. It is true, however there is a hiccup. The most simple case when we need to make an Hour from a readily available Integer is much harder than it seems at first.
> ...
> Surely there has to be an out of the box approach to fail gracefully from within a pure function and access the error message, similarly to how it was done in the parsing example.
> The first question that comes to mind when thinking about this problem is: "Why is there no MonadFail instance for Either String?" That would be an ideal solution, wouldn't it? But there is no such instance!

I didn't consider any of the advanced solutions to the problem suggested in the blog post, because I only have this problem once (so far). So I just added this ugly `unsafe` function. If this happens more frequently, then we need something better. And I'm absolutely open for suggestions :+1:

Reply:
OK, I played a bit around. While I did come up with something arguably slightly more elegant, it may also be more verbose, so happy to keep things as you have them. What I tried was this:

```Haskell
makeJannoCountryEither :: String -> Either PoseidonException JannoCountry
makeJannoCountry x =
    case decodeAlphaTwo (T.pack x) of
        Just c  -> Right $ JannoCountry c
        Nothing -> Left . PoseidonGenericException $ x ++ " is not a valid ISO-alpha2 code describing an existing country"

makeJannoCountryUnsafe :: String -> JannoCountry
makeJannoCountryUnsafe x = let Right r = makeJannoCountryEither x in r

makeJannoCountry :: (MonadFail m) => String -> m JannoCountry
makeJannoCountry x = case makeJannoCountry x of
    Left e -> fail . show $ e
    Right r -> return r
```

I think this would work, and it would define the error only once. But I can see that this is such a marginal improvement over your proposal that I wouldn't bother. Just wanted to share.

Reply:
Alright. I used your version :+1: